core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -8724,6 +8724,64 @@ protected void checkAggType(SqlTester tester, String expr, String type) {
         0d);
   }
 
+  @Test public void testBoolAndFunc() {
+    tester.setFor(SqlStdOperatorTable.BOOL_AND, VM_EXPAND);
+
+    tester.checkFails(
+        "bool_and(^*^)",
+        "Unknown identifier '\\*'",
+        false);
+    tester.checkType("bool_and(true)", "BOOLEAN");
+    tester.checkFails("^bool_and(1)^",
+        "Cannot apply 'BOOL_AND' to arguments of type 'BOOL_AND\\(<INTEGER>\\)'\\. "
+            + "Supported form\\(s\\): 'BOOL_AND\\(<BOOLEAN>\\)'",
+        false);
+    tester.checkFails(
+        "^bool_and()^",
+        "Invalid number of arguments to function 'BOOL_AND'. Was expecting 1 arguments",
+        false);
+    tester.checkFails(
+        "^bool_and(true, true)^",
+        "Invalid number of arguments to function 'BOOL_AND'. Was expecting 1 arguments",
+        false);
+
+    final String[] values1 = {"true", "true", "null"};
```

Comment:
Shall we add a test of `values = {null}` ?

Reply:
Thanks @jinxing64 's replies. Updated as you reviewed.